util/retry: TestRetryWithMaxDurationAndMaxRetries failed

util/retry.TestRetryWithMaxDurationAndMaxRetries [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20518333?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20518333?buildTab=artifacts#/) on master @ [1c99d43430f11cfaf845cbe7747f7b0a52686f4e](https://github.com/cockroachdb/cockroach/commits/1c99d43430f11cfaf845cbe7747f7b0a52686f4e):


```
=== RUN   TestRetryWithMaxDurationAndMaxRetries
--- FAIL: TestRetryWithMaxDurationAndMaxRetries (0.08s)
=== RUN   TestRetryWithMaxDurationAndMaxRetries/max_retries_hit_before_max_duration
    retry_test.go:582: 
        	Error Trace:	pkg/util/retry/retry_test.go:582
        	Error:      	Not equal: 
        	            	expected: 4
        	            	actual  : 2
        	Test:       	TestRetryWithMaxDurationAndMaxRetries/max_retries_hit_before_max_duration
        	Messages:   	expected 4 attempts, got 2
--- FAIL: TestRetryWithMaxDurationAndMaxRetries/max_retries_hit_before_max_duration (0.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRetryWithMaxDurationAndMaxRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54766

--- a/pkg/util/retry/retry_test.go
+++ b/pkg/util/retry/retry_test.go
@@ -566,7 +566,7 @@ func TestRetryWithMaxDurationAndMaxRetries(t *testing.T) {
 			InitialBackoff: time.Millisecond,
 			MaxBackoff:     time.Millisecond * 20,
 			Multiplier:     2,
-			MaxDuration:    time.Millisecond * 50,
+			MaxDuration:    time.Second,
 			MaxRetries:     3,
 		}
 

